OfflineService/OfflineServiceViewController.swift

```diff
@@ -160,51 +156,73 @@ class OfflineServiceViewController: UITableViewController, OfflineServiceDataSou
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         tableView.deselectRow(at: indexPath, animated: true)
         
-        let offlineDataRegion = offlineDataItems[indexPath.row]
-        showActions(offlineDataRegion)
+        let offlineRegion = offlineRegions[indexPath.row]
+        showActions(offlineRegion)
     }
     
     // MARK: - Private methods
 
-    private func showActions(_ offlineDataRegion: OfflineDataItem) {
+    private func showActions(_ offlineRegion: OfflineRegion) {
```

Comment:
After introducing new logic it's no longer possible to correctly remove region in some cases: e.g. action which allows to delete region appears before fully downloading region, region also remains as downloaded. Can you check previous behavior to mimic it?

Reply:
fixed